python/tests/test_umap.py

```diff
@@ -385,15 +394,16 @@ def test_umap_sample_fraction(gpu_number: int) -> None:
         pyspark_type = "float"
         feature_cols = [f"c{i}" for i in range(X.shape[1])]
         schema = [f"{c} {pyspark_type}" for c in feature_cols]
-        df = spark.createDataFrame(X.tolist(), ",".join(schema))
+        df = spark.createDataFrame(X.tolist(), ",".join(schema)).coalesce(1)
```

Comment:
Nit: do we need coalesce here or it is optional?

Reply:
Mainly a peace of mind thing to ensure determinism, such that the sampling is applied to a single partition; though it would only really be needed if there were upstream joins/shuffles that could introduce [non-deterministic behavior](https://medium.com/udemy-engineering/pyspark-under-the-hood-randomsplit-and-sample-inconsistencies-examined-7c6ec62644bc).